### DISCUSSION TITLE
Lock scroll on html for modals/drawers

### DISCUSSION THREAD

**Discussion by fernandodof** [chakra-ui/chakra-ui#8672] [General]
### Issue Summary

When a modal is opened, it locks the body scroll of the page. However, if the scroll is applied to the HTML element instead of the body element, the scroll is not locked, and the page is still scrollable. I would like to have an option to lock the HTML scroll as well when a modal is open.

### Problem Statement/Justification

The current behavior is the HTML scroll remains active when a modal is open. Users can unintentionally scroll the background content, causing confusion and disrupting the intended focus on the modal.

## Possible Solution

Lock scroll on both body and HTML by default and have the option to change it.

```
<Modal isOpen={isOpen} onClose={onClose} lockScrollOn={['html', 'body']}>
   
</Modal>
```

### Alternatives

Some things that I have considered:
- Overriding the modal behavior lock the HTML scroll 
- Adding a CSS rule to lock the scroll on the HTML if the body has `data-scroll-locked`
```html
html:has(body[data-scroll-locked]) {
    overflow: hidden;
}
```

**Comment by segunadebayo** [comment]
`Modal` is v2. Put page scroll on `body`, not `html`.

In v3, `Dialog.Root` has `preventScroll`. If you must scroll `html`, lock it with `html:has([data-scroll-lock]) { overflow: hidden }`.
